cura/NetworkClient.py

```diff
@@ -0,0 +1,220 @@
+# Copyright (c) 2018 Ultimaker B.V.
+# Cura is released under the terms of the LGPLv3 or higher.
+from time import time
+from typing import Optional, Dict, Callable, List
+
+from PyQt5.QtCore import QUrl
+from PyQt5.QtNetwork import QNetworkAccessManager, QNetworkReply, QHttpMultiPart, QNetworkRequest, QHttpPart, \
+    QAuthenticator
+
+from UM.Logger import Logger
+from cura.CuraApplication import CuraApplication
+
+
+##  Abstraction of QNetworkAccessManager for easier networking in Cura.
+#   This was originally part of NetworkedPrinterOutputDevice but was moved out for re-use in other classes.
+class NetworkClient:
+    
+    def __init__(self, application: CuraApplication = None):
```

Comment:
Missing return type. It's also not needed to pass the application in the constructor, it's a singleton.